packages/components/src/presentation-components/instance-filter-builder/Utils.ts

```diff
@@ -221,5 +221,24 @@ function isPropertyNumeric(typename: string) {
 }
 
 function isInvalidNumericValue(value: PrimitiveValue) {
-  return value.displayValue !== undefined && value.displayValue !== "" && isNaN(Number(value.value));
+  if (!value.displayValue || !value.value) {
+    return true;
+  }
+
+  if (typeof value.value !== "string") {
+    return isNaN(Number(value.value));
+  }
+
+  const deserialized = deserializeDisplayValueGroupArray(value.displayValue, value.value);
+
+  let isInvalid = false;
+  deserialized.groupedRawValues &&
+    deserialized.groupedRawValues.forEach((element: Value[]) => {
+      if (isNaN(Number(element))) {
+        isInvalid = true;
+        return;
+      }
+    });
+
+  return deserialized.groupedRawValues && isInvalid;
```

Comment:
This is handling our hack for implementing the `IN` operator without introducing braking changes into the `components-react` package by setting the value to a serialized JSON object. Please add a comment here explaining that.